shiksha-api/app-service/tests/unit/observability/test_question_bank_trace.py

```
@@ -0,0 +1,47 @@
+
+    import app.services.question_paper_service  # ensure imported before patch resolves
+    with patch("app.services.question_paper_service.get_client", return_value=mock_lf):
+        from app.services.question_paper_service import QuestionPaperService
+        svc = QuestionPaperService.__new__(QuestionPaperService)
+        svc.client = MagicMock()
+        svc._adapter_cache = {}
+
+        try:
+            await svc.generate_question_bank_by_parts(req)
+        except Exception:
+            pass  # we only care that trace was set up before any downstream error
```

Ruff (0.15.18)

[warning] Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

(SIM105)